### ISSUE TITLE
Exclude isn't working if extension in include path

### ISSUE THREAD

**Issue by AliceSimon** [ansible/ansible-lint#3635]
##### Summary

Up to version `6.16.1`, I can include a type of file and exclude a file that I don't want to analyze with ansible-lint.

But with follow version, it's not working anymore.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.17.3.dev12 using ansible-core:2.15.2 ansible-compat:4.1.5 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
You are using a pre-release version of ansible-lint.
```

##### STEPS TO REPRODUCE

1. Create three two Ansible files with yml extension and one with yaml:
![image](https://github.com/ansible/ansible-lint/assets/121865194/1359f436-7140-47f1-92c3-ade8600542da)
2. Try to do an analysis only on yml file and exclude one of them:

```console (paste below)
ansible-lint roles/*.yml --exclude roles/roles.yml
```

##### Desired Behavior

I can exclude a file and only include yml files:

![image](https://github.com/ansible/ansible-lint/assets/121865194/fa190cf7-9031-415d-9787-0c836b17d8ae)

##### Actual Behavior

Yaml file is well ignore but exclude isn't working:

![image](https://github.com/ansible/ansible-lint/assets/121865194/8ea1a6d7-9dbf-4667-a5d8-71ddb4376c2b)

**Comment by AliceSimon** [comment]
@ajinkyau, sure.

```bash
ansible-lint roles/ --exclude roles/roles.yml
WARNING  Listing 10 violation(s) that are fatal
fqcn[action-core]: Use FQCN for builtin module actions (apt).
roles/roles2.yaml:5 Use `ansible.builtin.apt` or `ansible.legacy.apt` instead.

fqcn[action-core]: Use FQCN for builtin module actions (service).
roles/roles2.yaml:10 Use `ansible.builtin.service` or `ansible.legacy.service` instead.  

fqcn[action-core]: Use FQCN for builtin module actions (template).
roles/roles2.yaml:16 Use `ansible.builtin.template` or `ansible.legacy.template` instead.

risky-file-permissions: File permissions unset or incorrect.
roles/roles2.yaml:16 Task/Handler: Copy Nginx configuration

yaml[new-line-at-end-of-file]: No new line character at the end of file
roles/roles2.yaml:21

fqcn[action-core]: Use FQCN for builtin module actions (apt).
roles/roles2.yml:5 Use `ansible.builtin.apt` or `ansible.legacy.apt` instead.

fqcn[action-core]: Use FQCN for builtin module actions (service).
roles/roles2.yml:10 Use `ansible.builtin.service` or `ansible.legacy.service` instead.

fqcn[action-core]: Use FQCN for builtin module actions (template).
roles/roles2.yml:16 Use `ansible.builtin.template` or `ansible.legacy.template` instead.

risky-file-permissions: File permissions unset or incorrect.
roles/roles2.yml:16 Task/Handler: Copy Nginx configuration

yaml[new-line-at-end-of-file]: No new line character at the end of file
roles/roles2.yml:21

Read documentation for instructions on how to ignore specific rule violations.

                       Rule Violation Summary
 count tag                           profile    rule associated tags 
     2 yaml[new-line-at-end-of-file] basic      formatting, yaml     
     2 risky-file-permissions        safety     unpredictability     
     6 fqcn[action-core]             production formatting

Failed: 10 failure(s), 0 warning(s) on 4 files. Last profile that met the validation criteria was 'min'.
```
With those files:
![image](https://github.com/ansible/ansible-lint/assets/121865194/ba942bc1-0f2a-4599-ba54-b27551a08c52)

**Comment by ssbarnea** [comment]
This bug does not reproduce with current version:

```
$ al -vv roles/ --exclude roles/3.yml
DEBUG    Folder `roles` does not look like a role due to missing any of the common subfolders such: tasks, meta, vars, defaults, handlers.
INFO     Excluded: roles/3.yml
DEBUG    Passed 'roles/1.yml' positional argument was identified as generic 'yaml' file kind.
DEBUG    Passed 'roles/2.yml' positional argument was identified as generic 'yaml' file kind.

DEBUG    Examining roles/1.yml of type yaml
DEBUG    Examining roles/2.yml of type yaml
```

**Comment by AliceSimon** [comment]
Hello @ssbarnea,

I think I still have the issue with this version:
![image](https://github.com/ansible/ansible-lint/assets/121865194/81025990-8b48-4eaa-b404-c54ee20bd8b3)

For instance:
`ansible-lint plays/*yml --exclude plays/launchpad-foundation.yml `

![image](https://github.com/ansible/ansible-lint/assets/121865194/5658d4c8-313a-4567-9f9e-7d1d3f0849fc)
